include/openPMD/IO/AbstractIOHandlerImpl.hpp

```diff
@@ -123,7 +123,7 @@ class AbstractIOHandlerImpl
                         availableChunks(i.writable, deref_dynamic_cast< Parameter< O::AVAILABLE_CHUNKS > >(i.parameter.get()));
                         break;
                 }
-            } catch (unsupported_data_error&)
+            } catch (...)
             {
                 (*m_handler).m_work.pop();
```

Comment:
Should we write a warning here? I think this catch should not happen in a well-defined program and could potentially complicate debugging later on if we do this silently.

Reply:
Yep, probably a good idea and useful as a starting point for debugging. 
(Also, the current implementation has led to some quite painful debugging experiences for me, because the failing task is tried *again* upon destruction, leading to very weird error behavior. Making this change is the right call imo)

Reply:
Another note: This code is not used by the ADIOS1 backend. Should we add the same logic there?

Reply:
Pushed an update, current output of the test:
```
Filters: automatically_deactivate_span
[AbstractIOHandlerImpl] IO Task GET_BUFFER_VIEW failed with exception. Removing task from IO queue and passing on the exception.
Ignoring expected error: ERROR: span currently not supported with operations, remove calls to Variable<T>::AddOperation or use Variable<T>::RemoveOperations, in call to Variable<T>::Span Engine::Put
```

Note that this only removes the failing task from the IO queue but no others. We don't really do any error recovery at the moment, we already talked about improving that aspect in the future. When we do this, it will probably show which implementation makes more sense (removing only the failing task or clearing the queue).

Reply:
> Another note: This code is not used by the ADIOS1 backend. Should we add the same logic there?

Yes, good idea. Do you like to push that?